### PR TITLE
Fixed a bunch of FindBugs warning

### PR DESCRIPTION
**Pull request by liyinan926**
Signed-off-by: Yinan Li <liyinan926@gmail.com>

### CODE REVIEW

**Comment by liyinan926** [comment]
@zliu41 can you review?

File: gobblin-runtime/src/main/java/gobblin/runtime/util/JobStateToJsonConverter.java [hunk 1]
```diff
@@ -223,8 +224,11 @@ public static void main(String[] args)
...
       Closer closer = Closer.create();
       try {
-        FileWriter fw = closer.register(new FileWriter(cmd.getOptionValue('t')));
-        fw.write(stringWriter.toString());
+        FileOutputStream fileOutputStream = closer.register(new FileOutputStream(cmd.getOptionValue('t')));
+        OutputStreamWriter outputStreamWriter =
+            closer.register(new OutputStreamWriter(fileOutputStream, ConfigurationKeys.DEFAULT_CHARSET_ENCODING));
+        BufferedWriter bufferedWriter = closer.register(new BufferedWriter(outputStreamWriter));
+        bufferedWriter.write(stringWriter.toString());
       } finally {
```
Comment (zliu41) [inline gobblin-runtime/src/main/java/gobblin/runtime/util/JobStateToJsonConverter.java]: Should rethrow Throwable before finally?

**Comment by zliu41** [comment]
LGTM with 1 comment

File: gobblin-runtime/src/main/java/gobblin/runtime/util/JobStateToJsonConverter.java (hunk 1 above)
Comment (liyinan926) [inline gobblin-runtime/src/main/java/gobblin/runtime/util/JobStateToJsonConverter.java]: Fixed.

**Pull request merged**